base/Data/Universe/Class.hs

```diff
@@ -14,6 +14,13 @@ import Data.Universe.Helpers
 -- recursively enumerable (and that 'universe' is that enumeration). In
 -- particular, you promise that any finite inhabitant has a finite index in
 -- 'universe', and that no inhabitant appears at two different finite indices.
+--
+-- /Laws:/
+--
+-- @
+-- 'elem' x 'universe'                       -- any inhabitant has a finite index
+-- 'length' ('filter' (== x) 'universe') == 1  -- if terminates
```

Comment:
This doesn't seem quite strong enough: it should also be the case that `filter (x==) universe` never produces `x:y:_`.